test/org/jitsi/impl/neomedia/transform/fec/FlexFec03ReceiverTest.java

```diff
@@ -0,0 +1,153 @@
+package org.jitsi.impl.neomedia.transform.fec;
+
+import org.easymock.*;
+import org.jitsi.service.configuration.*;
+import org.jitsi.service.libjitsi.*;
+import org.jitsi.service.neomedia.*;
+import org.junit.*;
+import org.junit.runner.*;
+import org.powermock.api.easymock.*;
+import org.powermock.core.classloader.annotations.*;
+import org.powermock.modules.junit4.*;
+
+import java.util.*;
+
+import static org.easymock.EasyMock.expect;
+import static org.junit.Assert.*;
+import static org.powermock.api.easymock.PowerMock.*;
+
+/**
+ * Created by bbaldino on 11/15/17.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(LibJitsi.class)
+public class FlexFec03ReceiverTest
+{
+    private ConfigurationService mockConfigurationService;
+
+    @Before
+    public void setUp()
+    {
+        PowerMock.mockStatic(LibJitsi.class);
+        mockConfigurationService = PowerMock.createMock(ConfigurationService.class);
+        expect(mockConfigurationService.getInt(EasyMock.anyString(), EasyMock.anyInt())).andAnswer(new IAnswer<Integer>() {
+            public Integer answer()
+            {
+                return (Integer)EasyMock.getCurrentArguments()[1];
+            }
+
+        }).anyTimes();
+        expect(LibJitsi.getConfigurationService()).andReturn(mockConfigurationService).anyTimes();
+    }
+
+    @After
+    public void tearDown()
+    {
+        verifyAll();
+    }
+
+    public class FecCaptureReadResult
+    {
+        Set<FlexFec03Packet> flexFecPackets = new HashSet<>();
+        Map<Integer, RawPacket> mediaPackets;
+    }
+
+    /**
+     * Returns the RawPacket whose sequence number matches recoveredPacketSeqNum
+     * if it exists in the given RawPacket array
+     * @param packets the packets to search
+     * @param recoveredPacketSeqNum the sequence number of the desired packet
+     * @return the RawPacket in packets whose sequence number matches
+     * recoveredPacketSeqNum or null if it isn't present
+     */
+    private RawPacket getRecoveredPacket(RawPacket[] packets, int recoveredPacketSeqNum)
+    {
+        for (RawPacket packet : packets)
+        {
+            if (packet.getSequenceNumber() == recoveredPacketSeqNum)
+            {
+                return packet;
+            }
+        }
+        return null;
+    }
+
+    private void verifyFlexFec(FlexFec03Packet flexFecPacket, Map<Integer, RawPacket> mediaPackets)
+    {
+        List<Integer> protectedSeqNums = flexFecPacket.getProtectedSequenceNumbers();
+        List<RawPacket> protectedMediaPackets = new ArrayList<>();
+        for (Integer protectedSeqNum : protectedSeqNums)
+        {
+            protectedMediaPackets.add(mediaPackets.get(protectedSeqNum));
+        }
+        // We'll treat the first one as the 'lost' packet we'll expect the
+        // FlexFec03Receiver to recover
```

Comment:
I think it would be nice to test the case of different packets being lost (not just the first one). Should be easy to do with Random with a fixed seed

Reply:
i'm fine with that, as long as the test itself is consistent (i.e. we either always lose the same packet or run through every scenario, each where a different one of the protected packets is lost).

Reply:
i changed it to test dropping all possible media packets (one by one).  doesn't seem to add a noticeable time difference.